Mathlib/AlgebraicGeometry/Stalk.lean

```diff
@@ -51,12 +61,264 @@ theorem IsAffineOpen.fromSpecStalk_eq {X : Scheme} (x : X) {U V : X.Opens}
 If `x` is a point of `X`, this is the canonical morphism from `Spec(O_x)` to `X`.
 -/
 noncomputable def Scheme.fromSpecStalk (X : Scheme) (x : X) :
-    Scheme.Spec.obj (op (X.presheaf.stalk x)) ⟶ X :=
+    Spec (X.presheaf.stalk x) ⟶ X :=
   (isAffineOpen_opensRange (X.affineOpenCover.map x)).fromSpecStalk (X.affineOpenCover.covers x)
 
 @[simp]
-theorem IsAffineOpen.fromSpecStalk_eq_fromSpecStalk
-    {X : Scheme} {U : X.Opens} (hU : IsAffineOpen U) {x : X} (hxU : x ∈ U) :
+theorem IsAffineOpen.fromSpecStalk_eq_fromSpecStalk {x : X} (hxU : x ∈ U) :
     hU.fromSpecStalk hxU = X.fromSpecStalk x := fromSpecStalk_eq ..
 
+lemma IsAffineOpen.fromSpecStalk_closedPoint {U : Opens X} (hU : IsAffineOpen U)
+    {x : X} (hxU : x ∈ U) :
+    (hU.fromSpecStalk hxU).1.base (closedPoint (X.presheaf.stalk x)) = x := by
+  rw [IsAffineOpen.fromSpecStalk, Scheme.comp_val_base_apply]
+  erw [← hU.primeIdealOf_eq_map_closedPoint ⟨x, hxU⟩, hU.fromSpec_primeIdealOf ⟨x, hxU⟩]
+
+namespace Scheme
+
+@[simp]
+lemma fromSpecStalk_closedPoint {x : X} :
+    (X.fromSpecStalk x).1.base (closedPoint (X.presheaf.stalk x)) = x :=
+  IsAffineOpen.fromSpecStalk_closedPoint _ _
+
+lemma fromSpecStalk_app {x : X} (hxU : x ∈ U) :
+    (X.fromSpecStalk x).app U =
+      X.presheaf.germ U x hxU ≫
+        (ΓSpecIso (X.presheaf.stalk x)).inv ≫
+          (Spec (X.presheaf.stalk x)).presheaf.map (homOfLE le_top).op := by
+  obtain ⟨_, ⟨V : X.Opens, hV, rfl⟩, hxV, hVU⟩ := (isBasis_affine_open X).exists_subset_of_mem_open
+    hxU U.2
+  rw [← hV.fromSpecStalk_eq_fromSpecStalk hxV, IsAffineOpen.fromSpecStalk, Scheme.comp_app,
+    IsAffineOpen.fromSpec_app_of_le hV _ hVU, ←  X.presheaf.germ_res (homOfLE hVU) x hxV]
+  simp only [Category.assoc]
+  rw [Hom.naturality, ← ΓSpecIso_inv_naturality_assoc]
+  rfl
```

Comment:
```suggestion
  rw [← hV.fromSpecStalk_eq_fromSpecStalk hxV, IsAffineOpen.fromSpecStalk, Scheme.comp_app,
    hV.fromSpec_app_of_le _ hVU, ← X.presheaf.germ_res (homOfLE hVU) x hxV]
  simp [Category.assoc, ← ΓSpecIso_inv_naturality_assoc]
```

Reply:
Hmm this doesn't work for me.

Reply:
This is strange, I just copy pasted this into your new version and it still works for me. Probably it is not very important, I just like that we can get rid of the terminal `rfl`.

Reply:
Oh I thought it was `simp only`. Thanks!